tests/test_sharing_client.py

```diff
@@ -0,0 +1,229 @@
+import unittest
+from unittest.mock import patch
+
+from test_utils import *
+from uid2_client import SharingClient, ClientType, EncryptionError, encryption
+
+
+@patch('uid2_client.sharing_client.refresh_sharing_keys')
+class TestSharingClient(unittest.TestCase):
+    _CONST_BASE_URL = 'base_url'
+    _CONST_API_KEY = 'api_key'
+
+    def setUp(self):
+        self._key_collection = create_key_collection(IdentityScope.UID2)
+        self._client = SharingClient(self._CONST_BASE_URL, self._CONST_API_KEY, client_secret)
+
+    def test_smoke_test(self, mock_refresh_sharing_keys):  # SmokeTest
+        for expected_scope, expected_version in test_cases_all_scopes_all_versions:
+            with self.subTest(expected_scope=expected_scope, expected_version=expected_version):
+                token = generate_uid_token(expected_scope, expected_version)
+                mock_refresh_sharing_keys.return_value = create_key_collection(expected_scope)
+                self._client.refresh_keys()
+                decrypted = self._client.decrypt_token_into_raw_uid(token)
+                self.assertEqual(decrypted.identity_scope, expected_scope)
```

Comment:
similar to bidstream tests, seems we're missing the raw uid comparisons in 3 tests here, let's make a `DecryptAndAssertSuccess` function to make it less error prone and easier for the reader

Reply:
Done

Reply:
looks good although I think we have some unnecessary `self.assertTrue(refresh_response.success)` remaining

Reply:
I think its good to have and its also present in C# SDk. https://github.com/IABTechLab/uid2-client-python/pull/31/files#diff-ed391f74a93f020690d1d8657357dfbcab403e1f24d79f02fe0a0343490c7b97R98

Reply:
I'm not following, isn't `decrypt_and_assert_success` calling `self.assertTrue(refresh_response.success)` already? Why do we need to duplicate that assert?

Reply:
ah never mind, I was confusing refresh success with decrypt success